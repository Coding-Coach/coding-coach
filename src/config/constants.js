export default {
  contact: {
    EMAIL: process.env.REACT_APP_EMAIL || 'codingcoachio@gmail.com',
    SLACK_URL:
      process.env.REACT_APP_SLACK_URL ||
      'https://join.slack.com/t/coding-coach/shared_invite/enQtNDMyMzUzNjAxODQyLTUwYTFkMzZmNGFhMzZjYTQwOWE2YWFjOGJhNzYyOGIxNDM5Zjc1YWQ4NjMwN2U5YzBlNjkwMTI2ZDNiOWQyMzM',
  },
  social: {
    FB_URL: process.env.REACT_APP_FACEBOOK_URL || 'https://www.facebook.com/codingcoachio/',
    INSTA_URL: process.env.REACT_APP_INSTA_URL || 'https://www.instagram.com/codingcoach_io/',
    TWITTER_URL: process.env.REACT_APP_TWITTER_URL || 'https://twitter.com/codingcoach_io',
  },
  legal: {
    PRIVACY_POLICY_URL:
      process.env.REACT_APP_PRIVACY_POLICY_URL ||
      'https://www.freeprivacypolicy.com/free-privacy-policy-generator.php',
  },
  payments: {
    paypal: process.env.REACT_APP_PAYPAL_DONATE_BUTTON_ID || '123abc',
  },
};
