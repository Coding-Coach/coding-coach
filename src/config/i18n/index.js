import en from './en/messages.js';
import es from './es/messages.js';
import fr from './fr/messages.js';

export const catalogs = { en, es, fr };
