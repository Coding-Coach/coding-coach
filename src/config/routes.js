const routesConfiguration = {
  landing: {
    name: 'landing_home',
    home: '/',
  },
  dashboard: {
    name: 'home',
    group: 'dashboard',
    path: '/app/dashboard',
  },
  messages: {
    name: 'messages',
    group: 'messages',
    path: '/app/messages',
  },
  mentorships: {
    name: 'mentorships',
    group: 'mentorships',
    path: '/app/mentorships',
  },
  settings: {
    name: 'settings',
    group: 'settings',
    path: '/app/settings',
  },
};

export { routesConfiguration };
