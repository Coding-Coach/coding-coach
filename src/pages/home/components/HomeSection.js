import React from 'react';
import styles from './HomeSection.module.scss';
import PropTypes from 'prop-types';

const HomeSection = ({ id, title, text, media, children }) => (
  <section id={id} className={styles.section}>
    <div className={styles.sectionInner}>
      <div>
        <header className={styles.header}>
          <h1 className={styles.title}>{title}</h1>
        </header>
        {styles.description && <p className={styles.description}>{text}</p>}
        {children}
      </div>
      {media}
    </div>
  </section>
);

HomeSection.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  text: PropTypes.string,
  media: PropTypes.node.isRequired,
  children: PropTypes.node,
};

export default HomeSection;
