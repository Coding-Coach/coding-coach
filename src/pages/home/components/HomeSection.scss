@import '~theme/base/variables';
@import '~theme/base/mediaqueries';
@import '~theme/base/utils';

$first-flex-element-width: 50%;
$second-flex-element-width: 45%;
$space-between-text-image: 50px;

.section {
    padding: 75px 30px;
    position: relative;

    @media screen and (min-width: 768px) {
        padding: 100px 0;
    }
}

.sectionInner {
    align-items: center;
    display: flex;
    flex-direction: column;
    width: 100%;

    // Section Title and text
    > *:nth-child(1) {
        margin-bottom: $space-between-text-image;
    }
    // Media
    > *:nth-child(2) {
        max-width: 100%;
    }

    @media screen and (min-width: 768px) {
        @include generalAlignment();
        margin: 0 auto;
        flex-direction: row;
        justify-content: space-between;

        > *:nth-child(1) {
            margin-bottom: 0;
            width: $first-flex-element-width;
        }

        > *:nth-child(2) {
            width: $second-flex-element-width;
        }
    }
}

.header {
    margin-bottom: 1.25rem;
    position: relative;

    &:after {
        content: '';
        background: $color-grey-light;
        bottom: 0;
        position: absolute;
        height: 1px;
        width: 250px;
    }
}

.title {
    color: $color-grey-dark;
    text-transform: uppercase;
    font-family: $font-secondary;
    font-size: $font-xxl;
    margin: 0;
    line-height: 1.25;
    text-transform: uppercase;
    font-weight: normal;
}

.description {
    color: $color-grey-dark;
    font-size: $font-s;
    font-family: $font-primary;
    line-height: 1.5;
    margin: 0;

    @media screen and (min-width: 1024px) {
        font-size: $font-m;
    }
}