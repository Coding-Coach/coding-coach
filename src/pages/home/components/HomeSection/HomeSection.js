import React from 'react';
import styles from './HomeSection.scss';
import PropTypes from 'prop-types';

const HomeSection = ({ id, title, text, media, children }) => (
  <section id={id} className={styles.section}>
    <div className={styles.sectionInner}>
      <div>
        <header className={styles.header}>
          <h1 className={styles.title}>{title}</h1>
        </header>
        <p className={styles.description}>{text}</p>
        {children}
      </div>
      {media}
    </div>
  </section>
);

HomeSection.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  media: PropTypes.node.isRequired,
  children: PropTypes.node,
};

export default HomeSection;
