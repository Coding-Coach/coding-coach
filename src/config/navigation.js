import { routesConfiguration } from 'config/routes';
import IconDashboard from 'components/icon/Dashboard';
import IconMessages from 'components/icon/Messages';
import IconMentorships from 'components/icon/Mentorships';
import IconSettings from 'components/icon/Settings';

const navConfiguration = {
  tabs: [
    {
      name: routesConfiguration.dashboard.name,
      path: routesConfiguration.dashboard.path,
      group: routesConfiguration.dashboard.group,
      label: 'dashboard.navigation.tab.home',
      icon: IconDashboard,
    },
    {
      name: routesConfiguration.messages.name,
      path: routesConfiguration.messages.path,
      group: routesConfiguration.messages.group,
      label: 'dashboard.navigation.tab.messages',
      icon: IconMessages,
    },
    {
      name: routesConfiguration.mentorships.name,
      path: routesConfiguration.mentorships.path,
      group: routesConfiguration.mentorships.group,
      label: 'dashboard.navigation.tab.mentorships',
      icon: IconMentorships,
    },
    {
      name: routesConfiguration.settings.name,
      path: routesConfiguration.settings.path,
      group: routesConfiguration.settings.group,
      label: 'dashboard.navigation.tab.settings',
      icon: IconSettings,
    },
  ],
};

export { navConfiguration };
