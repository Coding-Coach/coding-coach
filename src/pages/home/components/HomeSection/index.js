import HomeSection from './HomeSection';

export default HomeSection;
