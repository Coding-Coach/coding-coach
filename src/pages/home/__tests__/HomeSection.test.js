import React from 'react';
import { render } from 'react-testing-library';
import HomeSection from '../components/HomeSection';
import config from 'config/constants';

test('renders the home page section', () => {
  const { container } = render(<HomeSection />);
  expect(container.firstChild).toMatchSnapshot();
});
