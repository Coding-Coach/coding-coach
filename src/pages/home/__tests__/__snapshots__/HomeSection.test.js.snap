// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the home page section 1`] = `
<section>
  <div>
    <div>
      <header>
        <h1 />
      </header>
    </div>
  </div>
</section>
`;
